main.go

```diff
@@ -43,6 +43,10 @@ func main() {
 			Name:  "namespace, n",
 			Usage: "Set namespace. Uses the namespace set in the context by default",
 		},
+		cli.StringSliceFlag{
+			Name:  "release-name, r",
```

Comment:
Sorry but honestly saying, I'm not yet sure if this is the right name(if there's one) which describes the feature.
More concretely, I imagine that one would misunderstand this as a flag used for somehow overriding release names, which isn't true of course.

Instead, could this be any of the followings?

1. `helmfile sync --only myrelease1,myrelease2`
   - not very intuitive as it isn't clear that what we can filter & by what we can filter releases
2. `helmfile sync --only-releases myrelease1,myrelease2`
   - it isn't clear that by what we can filter releases
3. `helmfile sync --filter-releases myrelease1,myrelease2`
   - same as 3
4. `helmfile sync --filter-releases-by-names myrelease1,myrelease2`
   - imho, too verbose
5. `helmfile sync --filter-releases name=myrelease1,myrelease2`
   - what if we wanted to support #28? `--filter-releases tag=tier1:frontend`? not sure i could correctly differentiate `=` and `:` :)
6. `helmfile sync myrelease1,myrelease2`
7. `helmfile sync myrelease1 myrelease2`

I slightly prefer 7 as it straightly maps to my mental model that `helmfile` is working against releases in the first place. It would also look familiar to ones used to kubectl. `kubectl get po` works against all the pods, whereas `kubectl get po foo`  filters the target to a single pod.

A nice side-effect of it is that we now have a natural integration with shell brace expansion e.g. `helmfile sync myrelease{1,2}` 😃

Reply:
I would either recommend 7 or `helmfile sync --filter name=myrelease`  Where in the future we could support more filters

Reply:
`helmfile sync --filter name=myrelease` seems the most natural to me as well considering the proposed feature to use tags as a filter. 

7. is nice and clean but I don't think would be extended with a tag structure as well. It would be less obvious what the behavior would be with names and tags. `helmfile sync --tags tier:frontend myrelease1 myrelease2` would that apply the intersection or union of the two filters? This question stands either way but I think its more clear using a `--filter` flag that either tags or names should be used. 

I suppose 7 could lead to `helmfile sync tag:tier=frontend` or similar but that doesn't seem as clean to me

Reply:
@awwithro for supporting and/or something like
```helmfile sync --filter name=bob,tier=frontend --filter name=sam```
Any release that makes name=bob and tier=frontend AND any chart that has the name sam.

That would allow us to support and/or logic.

Reply:
@awwithro thanks that work will help toward the full tag/filter implementation

Reply:
@awwithro Thanks for your efforts! Please let us know if there's anything we can help. Probably questions regarding implementation, design, testing, etc?

Reply:
hey @mumoshu I've been a bit busy at work and haven't had a chance to circle back on this quite yet. I did have some questions though. I believe you're on the kubernetes slack? I can find you there